### PR TITLE
(Bluesky #1682) add OpenTelemetry tracing to status

### PR DESCRIPTION
**Pull request by dperl-dls**
To go along with https://github.com/bluesky/bluesky/pull/1724

  - Adds `opentelemetry-api` as a dependency
  - Registers a trace on `__init__` of `StatusBase` objects, ends it when they are marked done, and attaches some metadata to help identify what the status belongs to - this metadata is added for class specific metadata in each subclass
  - Registers traces on `StatusBase.wait()`

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -104,7 +104,7 @@
 # Example configuration for intersphinx: refer to the Python standard library.
 intersphinx_mapping = {
     "python": ("https://docs.python.org/3", None),
-    "bluesky": ("https://blueskyproject.io/bluesky/", None),
+    "bluesky": ("https://blueskyproject.io/bluesky/main", None),
```
Comment (mrakitin) [inline docs/conf.py:107]: Good update to get https://blueskyproject.io/bluesky/main/objects.inv.

**Review by mrakitin — COMMENTED** [review]
Looks like the `SubscriptionStatus` and `DeviceStatus` (and their subclasses) are not included. Was it intentional?

Why not to apply the changes to the `StatusBase` class?

**Comment by dperl-dls** [comment]
Hi @mrakitin , thanks for your review, sorry it took me so long to get back to you! The trace is indeed applied to `StatusBase`, I have updated the PR description to be more clear about this. For e.g. `DeviceStatus` the device is added as an attribute.

**Comment by mrakitin** [comment]
@dperl-dls, are the CI failures expected?

**Comment by dperl-dls** [comment]
@mrakitin yes, they're failing on master, unrelated to these changes. I can run all the tests locally and they pass, but I don't really know what I could do about the NSLS epics IOC containers. I was hoping this would be fixed by now and I could rebase.

**Comment by mrakitin** [comment]
> @dperl-dls, are the CI failures expected?

I think we need to refresh the image https://hub.docker.com/r/nsls2/epics-docker/tags to unblock the tests.

File: docs/conf.py (hunk 1 above)
Comment (mrakitin) [reply docs/conf.py:107]: Probably worth a separate PR.

**Review by danielballan — APPROVED** [review]
Looks straightforward.

**Review by mrakitin — APPROVED** [review]
Looks good to me!

**Pull request merged**